euth/process/models.py

```diff
@@ -0,0 +1,87 @@
+from django.db import models
+from django.core.urlresolvers import reverse
+from django.contrib.auth.models import Permission, User
+from django.utils import timezone
+
+
+class Process(models.Model):
+    name = models.CharField(max_length=128, db_index=True, unique=True)
+    title = models.CharField(max_length=1024)
+    description = models.TextField()
+    participants = models.ManyToManyField(
+        User,
+        related_name="process_participant",
+        blank=True,
+    )
+    moderators = models.ManyToManyField(
+        User,
+        related_name="process_moderator",
+    )
+
+    def get_current_phase(self):
+        now = timezone.now()
+        return self.phase_set\
+            .filter(
+                models.Q(date_start__lte=now, date_end__exact=None) |
+                models.Q(date_start__lte=now, date_end__gt=now))\
+            .first()
+
+    def get_absolute_url(self):
+        return reverse('process-detail', args=[str(self.name)])
+
+    def __str__(self):
+        return 'process «{}»'.format(self.name)
+
+
+class PhaseType(models.Model):
+    app_label = models.CharField(max_length=100)
+    name = models.CharField(max_length=100)
+
+    moderator_permissions = models.ManyToManyField(
+        Permission,
+        related_name="moderator_permissions",
+    )
+
+    participant_permissions = models.ManyToManyField(
+        Permission,
+        related_name="user_permissions",
+    )
+
+    def __str__(self):
+        return 'phase type «{}.{}»'.format(self.app_label, self.name)
+
+
+class ParticipationModule(models.Model):
+    pass
+
+
+class Phase(models.Model):
+
+    class Meta:
+        unique_together = (("process", "order"),)
+
+    title = models.CharField(max_length=1024)
+    description = models.TextField()
+
+    phase_type = models.ForeignKey(
+        PhaseType,
+        models.CASCADE,
+    )
+
+    order = models.PositiveSmallIntegerField(db_index=True)
+    process = models.ForeignKey(
+        Process,
+        models.CASCADE,
+    )
+
+    participation_module = models.ForeignKey(
+        ParticipationModule,
+        models.CASCADE,
+    )
+
+    date_start = models.DateTimeField(blank=True, null=True)
+    date_end = models.DateTimeField(blank=True, null=True)
+
+    def __str__(self):
+        return '{} phase of process «{}»'.format(
+            quantityStr(self.order), self.process.name)
```

Comment:
We need to add a validation to the phase checking that there is no other overlapping phase.